nix/overlays.nix

```diff
@@ -0,0 +1,9 @@
+# An attribute set containing arguments to be supplied to each overlay.
+args: 
+# The nixpkgs package set.
+pkgs:
+
+pkgs.lib.composeManyExtensions (map (f: f args) [
+  (import overlays/logict.nix)
```

Comment:
Replace these two with `overlays/haskell.nix` to make it easier to maintain.

Reply:
Haskell language server replaces `ghcid` and providing `cabal-install` in the shell can cause issues for GHC `9.4.2` and `9.2.4` in my experience.